src/query/server/server.go

```diff
@@ -196,17 +196,25 @@ func Run(runOpts RunOptions) {
 		}
 	}()
 
-	sigChan := make(chan os.Signal, 1)
-	signal.Notify(sigChan, os.Interrupt, syscall.SIGTERM)
-
 	var interruptCh <-chan error = make(chan error)
 	if runOpts.InterruptCh != nil {
 		interruptCh = runOpts.InterruptCh
 	}
 
-	select {
-	case <-sigChan:
-	case <-interruptCh:
+	if cfg.Backend == config.GRPCStorageType {
+		// Only use this if not running embedded, otherwise will
+		// obfuscate signal channel for m3db
+		sigChan := make(chan os.Signal, 1)
+		signal.Notify(sigChan, os.Interrupt, syscall.SIGTERM)
+
+		select {
+		case <-sigChan:
```

Comment:
nit: log the cause of termination so you can track why the process crashed

```
var interruptErr error
select {
case sig := <- sigChan:
  interruptErr = fmt.Errorf("%v", sig)
case err  := <- interruptCh:
  interruptErr = err
}

log.Infof("interrupt :%v", interruptErr)

Reply:
Good call